### DISCUSSION TITLE
Changing RX2 data-rate Value or from MAC

### DISCUSSION THREAD

**Discussion by daeynasvistas** [ngraziano/LMICPP-Arduino#29] [Q&A] [answered]
Do the LMICPP allways listen to RX2 data-rate SF12? can i change it .. or better, do join request/accept receive RX2 data-rate from NS server ? 

i think many people use TTN with LMICPP, is there some config to change the RX2 data-rate to SF9?

thanks.. :)

**Comment by ngraziano** [comment, marked as answer]
Lmicpp accept change of RX2 from the server during OTAA join, there is nothing to do.

For ABP there is setRx2Parameter see  [esp32-deepsleep-abp](/ngraziano/LMICPP-Arduino/blob/master/examples/esp32-deepsleep-abp/src/main.cpp)

**Reply by daeynasvistas** [reply]
great .. i love more and more this LMICPP every day
